js/components/main.js

```diff
@@ -1221,12 +1216,19 @@ class Main extends ImmutableComponent {
           key='tab-bar'
           activeFrameKey={activeFrame && activeFrame.get('key') || undefined}
           onMenu={this.onHamburgerMenu}
+          notificationBarActive={notificationBarOrigin}
           hasTabInFullScreen={
             sortedFrames
               .map((frame) => frame.get('isFullScreen'))
               .some(fullScreenMode => fullScreenMode === true)
           }
         />
+        {
+          this.props.appState.get('notifications') && this.props.appState.get('notifications').size && activeFrame
```

Comment:
We call out to `this.props.appState.get('notifications')` 3-4 times. Can we do this once, and then store the result?